owncloudApp/src/main/java/com/owncloud/android/ui/activity/DrawerActivity.kt

```diff
@@ -0,0 +1,699 @@
+        }
+    }
+
+    override fun onPostCreate(savedInstanceState: Bundle?) {
+        super.onPostCreate(savedInstanceState)
+        // Sync the toggle state after onRestoreInstanceState has occurred.
+        drawerToggle?.let {
+            it.syncState()
+            if (isDrawerOpen()) {
+                it.isDrawerIndicatorEnabled = true
+            }
+        }
+        updateAccountList()
+        updateQuota()
+    }
+
+    override fun onConfigurationChanged(newConfig: Configuration) {
+        super.onConfigurationChanged(newConfig)
+        drawerToggle?.onConfigurationChanged(newConfig)
+    }
+
+    override fun onBackPressed() {
+        if (isDrawerOpen()) {
+            closeDrawer()
+            return
+        }
+        super.onBackPressed()
+    }
+
+    override fun onResume() {
+        super.onResume()
+        setDrawerMenuItemChecked(checkedMenuItem)
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+
+        // update Account list and active account if Manage Account activity replies with
+        // - ACCOUNT_LIST_CHANGED = true
+        // - RESULT_OK
+        if (requestCode == ACTION_MANAGE_ACCOUNTS && resultCode == Activity.RESULT_OK && data!!.getBooleanExtra(
+                ManageAccountsActivity.KEY_ACCOUNT_LIST_CHANGED,
+                false
+            )
+        ) {
+
+            // current account has changed
+            if (data.getBooleanExtra(ManageAccountsActivity.KEY_CURRENT_ACCOUNT_CHANGED, false)) {
+                account = AccountUtils.getCurrentOwnCloudAccount(this)
+                // Refresh dependencies to be used in selected account
+                initDependencyInjection()
+                restart()
+            } else {
+                updateAccountList()
+                updateQuota()
+            }
+        }
+    }
+
+    /**
+     * Finds a view that was identified by the id attribute from the drawer header.
+     *
+     * @param id the view's id
+     * @return The view if found or `null` otherwise.
+     */
+    private fun findNavigationViewChildById(id: Int): View? {
+        return (findViewById<View>(R.id.nav_view) as NavigationView?)?.getHeaderView(0)?.findViewById(id)
+    }
+
+    override fun onAccountCreationSuccessful(future: AccountManagerFuture<Bundle?>?) {
+        super.onAccountCreationSuccessful(future)
+        updateAccountList()
+        updateQuota()
+        // Refresh dependencies to be used in selected account
+        initDependencyInjection()
+        restart()
+    }
+
+    /**
+     * populates the avatar drawer array with the first three ownCloud [Account]s while the first element is
+     * always the current account.
+     */
+    private fun populateDrawerOwnCloudAccounts() {
+        accountsWithAvatars = arrayOfNulls(3)
+        val accountsAll = AccountManager.get(this).getAccountsByType(accountType)
```

Comment:
Too much for a view... move as much logic as possible from activities and fragments to viewModel or utils